### ISSUE TITLE
function that has an infinite loop doesn't report anything

### ISSUE THREAD

**Issue by KotlinIsland** [DetachHead/basedpyright#804]
```py
def f() -> int:  # no error
    while True:
        print("hi")
```

this infinite loop only makes sense if the function is returning `Never`

**Comment by DetachHead** [comment]
the error should be on the `while True` saying the condition is always `True`, which is the same issue as #749

the reason you dont get an error on the return type is the same reason you don't get an error here, because `Never` is a subtype of `int`:
```py
def f() -> int:
    raise Exception()
```

**Comment by KotlinIsland** [comment]
`if True` is a mistake, `while True` is a common pattern, i would not expect an error on it

intentionally raising an exception to meet `int` makes practical sense, writing an infinite loop to meet `int` is nonsensical

this is a distinct issue to #749

**Comment by DetachHead** [comment]
i dont like the idea of special casing one of them when from a type checking perspective its the same thing

**Comment by KotlinIsland** [comment]
from a purely type checking perspective `1 + ""` returns `Never`, should we remove the diagnostic from this as well?

in all practicality, raising an exception is understandable and intentional. writing an infinite loop when you are supposed to return a value is guaranteed to be a mistake
